static/js/filter.js

```diff
@@ -36,43 +37,74 @@
             alias.style.display = "none";
         });
 
-        // BUG: If more than one labels are named the same thing (identical), 
-        // the filter only counts/shows the first result.
-		// Loop through both arrays, making that alias visible if it is a match.
+        // Fix GitHub/#966: Create temporary array of objects containing labels and their parent `.js-alias` DOM element
+        const searchIndexWithLabels = [];
+
+        for (const [index, alias] of aliasesWithLabelsCollection.entries()) {
```

Comment:
This is the crux of the PR: I join the both the array of label entries and each parent object together. When I filter it below, I can access (read: show/hide) the matched items.